externalLibrary/src/Main.elm

```diff
@@ -0,0 +1,316 @@
+
+        LinkExited exit ->
+            case exit of
+                Ok itemExit ->
+                  ( Debug.log "item" { model | stage = Exited, itemExit = Just itemExit }
+                  , Cmd.none
+                  )
+
+                Err err ->
+                  ( Debug.log "error :(" { model | stage = Exited }
+                  , Cmd.none
+                  )
+
+
+
+-- VIEW
+
+
+view : Model -> Html Msg
+view model =
+    div [] [ viewFor model ]
+
+
+viewFor : Model -> Html Msg
+viewFor model =
+    case model.stage of
+        Start ->
+            startView model
+
+        LinkAnotherBank ->
+            linkAnotherBankView model
+
+        Finish ->
+            finishView model
+
+        Exited ->
+            exitedView model
+
+
+startView : Model -> Html Msg
+startView model =
+    div []
+        [ 
+          div [] [ enrollingText model ]
+        , div [] [ button [ onClick <| OpenPlaidLink [ Auth ] ] [ text "Get Started" ] ]
+        ]
+
+
+
+linkAnotherBankView : Model -> Html Msg
+linkAnotherBankView model =
+    let
+        list =
+            banksLinkedSoFar model.items
+    in
+    div []
+        [ 
+          div [] [ enrollingText model ]
+        , div [] [ div [] [ h4 [] [ text "You've linked: " ] ]
+                 , div [] [ div [] [ ul [] list ] ]
+                 , div [] [ button [ onClick <| OpenPlaidLink [ Transactions ] ] [ text "Link another bank" ] ]
+                 , div [] [ button [ onClick ShowFinish ] [ text "Finish" ] ]
+                 ]
+        ]
+
+
+banksLinkedSoFar : List Item -> List (Html Msg)
+banksLinkedSoFar items =
+    List.concatMap
+        (\item ->
+            let
+                numberOfAccounts =
+                    item.accounts |> List.length |> String.fromInt
+
+                numberOfAccountsText =
+                    "- " ++ numberOfAccounts ++ " accounts"
+
+                accountNames =
+                    List.map .name item.accounts
+
+                accounts =
+                    String.join ", " accountNames
+            in
+            [ li [] [ text item.institution.name, span [ title accounts ] [ text numberOfAccountsText ] ]
+            ]
+        )
+        items
+
+
+finishView : Model -> Html Msg
+finishView model =
+    div []
+        [ h4 [] [ text "Thank you for linking your accounts." ]
+        , text "Here's where you might trigger an HTTP call."
+        ]
+
+
+exitedView : Model -> Html Msg
+exitedView model =
+    let
+        sessionId =
+          case model.itemExit of
+            Just ie ->
+              ie.metadata.sessionId
+
+            Nothing ->
+              "No sessionId"
+    in
+    div []
+        [ h4 [] [ text "Link Exited" ]
+        , text ("SessionId: " ++ sessionId)
+        ]
+
+
+enrollingText : Model -> Html Msg
+enrollingText model =
+    h3 [] [ "Enrolling " ++ model.name |> text ]
+
+
+institutionRowsForDepositorySelection : Model -> List (Html Msg)
+institutionRowsForDepositorySelection model =
```

Comment:
It looks like this and several other fuctions in `Main` are never used. I think it's worth either figuring out a way to use them in the example, or removing them entirely, otherwise they make it harder to find the more relevant parts of the code.